src/ucp/core/ucp_am.c

```diff
@@ -459,39 +470,46 @@ ucp_am_bcopy_pack_args_single(void *dest, void *arg)
 static size_t
 ucp_am_bcopy_pack_args_single_reply(void *dest, void *arg)
 {
-    ucp_am_reply_hdr_t *reply_hdr = dest;
-    ucp_request_t *req            = arg;
+    ucp_am_hdr_t *hdr  = dest;
+    ucp_request_t *req = arg;
+    ucp_am_reply_ftr_t *ftr;
     size_t length;
 
     ucs_assert(req->send.state.dt.offset == 0);
 
-    ucp_am_fill_header(&reply_hdr->super, req);
+    ucp_am_fill_header(hdr, req);
 
-    reply_hdr->ep_id  = ucp_send_request_get_ep_remote_id(req);
-    length            = ucp_am_bcopy_pack_data(reply_hdr + 1, req,
-                                               ucp_am_send_req_total_size(req));
+    length     = ucp_am_bcopy_pack_data(hdr + 1, req,
+                                        ucp_am_send_req_total_size(req));
+    ftr        = UCS_PTR_BYTE_OFFSET(hdr + 1, length);
+    ftr->ep_id = ucp_send_request_get_ep_remote_id(req);
 
     ucs_assert(length == ucp_am_send_req_total_size(req));
 
-    return sizeof(*reply_hdr) + length;
+    return sizeof(*hdr) + length + sizeof(*ftr);
 }
 
 static size_t
 ucp_am_bcopy_pack_args_first(void *dest, void *arg)
 {
-    ucp_am_first_hdr_t *hdr = dest;
-    ucp_request_t *req      = arg;
-    size_t length;
+    ucp_am_hdr_t *hdr  = dest;
+    ucp_request_t *req = arg;
+    ucp_am_first_ftr_t *first_ftr;
+    size_t length, max_length;
+
+    ucs_assert(req->send.state.dt.offset == 0);
 
-    length = ucs_min(ucp_am_send_req_total_size(req),
-                     ucp_ep_get_max_bcopy(req->send.ep, req->send.lane) -
-                     sizeof(*hdr));
+    ucp_am_fill_header(hdr, req);
 
-    ucp_am_fill_first_header(hdr, req);
+    max_length = ucs_min(ucp_am_send_req_total_size(req),
+                         ucp_ep_get_max_bcopy(req->send.ep, req->send.lane) -
+                                 UCP_AM_FIRST_FRAG_META_LEN);
```

Comment:
shout it be align as below?
```suggestion
max_length = ucs_min(ucp_am_send_req_total_size(req),
                      ucp_ep_get_max_bcopy(req->send.ep, req->send.lane) -
                      UCP_AM_FIRST_FRAG_META_LEN);
```

Reply:
clang likes this way, because it is a part of arithmetic operation, but not a new parameter